calico_containers/calico_ctl/utils.py

```diff
@@ -100,6 +101,56 @@ def get_container_ipv_from_arguments(arguments):
     return version
 
 
+def validate_cidr(argument):
+    """
+    Validate argument is in correct CIDR notation
+
+    :param argument: IP addr and associated routing prefix
+    :return: Boolean: True if valid IP, False if invalid
+    """
+    try:
+        netaddr.IPNetwork(argument)
+        return True
+    except (AddrFormatError, ValueError):
+    # Some versions of Netaddr have a bug causing them to return a
+    # ValueError rather than an AddrFormatError, so catch both.
+        return False
+
+
+def validate_ip(ip_addr, version):
+    """
+    Validate that ip_addr is a valid IPv4 or IPv6 address
+
+    :param ip_addr: valid IPv4 address
```

Comment:
This isn't necessarily a "valid" IP address, that's what we're testing.  Also, it could be IPv4 or IPv6.